### PR TITLE
Refactor DI startup to make Exception Replay can be started up alone

### PR DESCRIPTION
**Pull request by jpbempel**
# What Does This Do
Make sure Exception Replay can be started up without DI. Remote Config subscription is not started if DI is not enabled now. The debugger agent is initialized if either Dynamic Instrumentation or
 Exception Replay is enabled

# Motivation

# Additional Notes

# Contributor Checklist

- [ ] Format the title [according the contribution guidelines](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#title-format)
- [ ] Assign the `type:` and (`comp:` or `inst:`) labels in addition to [any usefull labels](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#labels)
- [ ] Squash your commits prior merging or merge using GitHub's [Squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits)
- [ ] Don't use `close`, `fix` or any [linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) when referencing an issue.  
  Use `solves` instead, and assign the PR [milestone](https://github.com/DataDog/dd-trace-java/milestones) to the issue
- [ ] Update the [public documentation](https://docs.datadoghq.com/tracing/trace_collection/library_config/java/) in case of new configuration flag or behavior

Jira ticket: [DEBUG-2735]

[DEBUG-2735]: https://datadoghq.atlassian.net/browse/DEBUG-2735?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dd-smoke-tests/debugger-integration-tests/src/test/java/datadog/smoketest/ExceptionDebuggerIntegrationTest.java [hunk 1]
```diff
@@ -34,6 +34,8 @@ protected ProcessBuilder createProcessBuilder(Path logFilePath, String... params
     commandParams.add("-Ddd.exception.replay.enabled=true"); // enable exception replay
     commandParams.add("-Ddd.internal.exception.replay.only.local.root=false"); // for all spans
     commandParams.add("-Ddd.third.party.excludes=datadog.smoketest");
+    // disable DI to make sure exception debugger works alone
+    commandParams.remove("-Ddd.dynamic.instrumentation.enabled=true");
```
Comment (shatzi) [inline dd-smoke-tests/debugger-integration-tests/src/test/java/datadog/smoketest/ExceptionDebuggerIntegrationTest.java:38]: false?

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/DebuggerAgent.java [hunk 2]
```diff
@@ -112,6 +108,36 @@ public static synchronized void run(
...
+    if (config.isDebuggerEnabled()) {
+      startDynamicInstrumentation(
+          instrumentation, sco, config, configurationUpdater, debuggerSink, classNameFiltering);
+    }
+    try {
+      /*
+      Note: shutdown hooks are tricky because JVM holds reference for them forever preventing
+      GC for anything that is reachable from it.
+       */
+      Runtime.getRuntime().addShutdownHook(new ShutdownHook(configurationPoller, debuggerSink));
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/DebuggerAgent.java:121]: configurationPoller can be null, is that a problem?

**Review by shatzi — APPROVED** [review]

File: dd-smoke-tests/debugger-integration-tests/src/test/java/datadog/smoketest/ExceptionDebuggerIntegrationTest.java (hunk 1 above)
Comment (jpbempel) [reply dd-smoke-tests/debugger-integration-tests/src/test/java/datadog/smoketest/ExceptionDebuggerIntegrationTest.java:38]: nop, we are **removing** the config added in the `getDebuggerParams()` method

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/DebuggerAgent.java (hunk 2 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/DebuggerAgent.java:121]: mot a problem, the weakReference will be null and by nature the weakRef is checked against null

**Pull request merged**